File: tests/data_processor_test.go
package main

import "testing"

func TestProcessData(t *testing.T) {
    result := ProcessData("test")
    if result == "" {
        t.Error("Expected non-empty result")
    }
}

func TestValidateInput(t *testing.T) {
    if !ValidateInput("valid") {
        t.Error("Expected true for valid input")
    }
}

func TestCalculateSum(t *testing.T) {
    result := CalculateSum(5, 10)
    if result != 15 {
        t.Errorf("Expected 15, got %d", result)
    }
}

func TestFormatOutput(t *testing.T) {
    result := FormatOutput("test", 123)
    if result == "" {
        t.Error("Expected non-empty result")
    }
}

File: tests/DataProcessorTest.kt
import org.junit.Test
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class DataProcessorTest {
    
    @Test
    fun testProcessData() {
        val result = processData("test")
        assertTrue(result.isNotEmpty())
    }
    
    @Test
    fun testValidateInput() {
        assertTrue(validateInput("valid"))
    }
    
    @Test
    fun testCalculateSum() {
        assertEquals(15, calculateSum(5, 10))
    }
    
    @Test
    fun testFormatOutput() {
        val result = formatOutput("test", 123)
        assertTrue(result.isNotEmpty())
    }
}
